frontend/src/modules/Dashboard/Components/Dashboard.tsx

```diff
@@ -127,6 +129,35 @@ export const Dashboard = () => {
     )
   }
 
+  // (a,b) = -1 if a before b, 1 if a after b, 0 if equal
+  function filtered(courseInfo: Course[], menuValue: FilterOption) {
+    switch (menuValue) {
+      case 'no-filter':
+        return courseInfo
+      case 'unmatchable':
+        return [...courseInfo].filter(
+          (course, _) =>
+            course.lastGroupNumber === 0 && course.unmatched.length === 1
+        )
+      case 'newly-matchable':
+        return [...courseInfo].filter(
+          (course, _) =>
+            course.lastGroupNumber === 0 && course.unmatched.length > 1
+        )
+      case 'matchable':
+        return [...courseInfo].filter(
+          (course, _) =>
+            (course.lastGroupNumber > 0 && course.unmatched.length > 0) ||
+            (course.lastGroupNumber === 0 && course.unmatched.length > 1)
+        )
+      case 'no-check-in-email':
```

Comment:
Nice separation of cases, the logic seems to work well. Did your tests cover these bottom cases here, where a match or email might be missing?

Reply:
Yup, I think I checked them.